cores/esp8266/Print.cpp

```diff
@@ -33,10 +33,20 @@
 
 /* default implementation: may be overridden */
 size_t Print::write(const uint8_t *buffer, size_t size) {
+
+#ifdef DEBUG_ESP_CORE
+    static char not_the_best_way [] ICACHE_RODATA_ATTR STORE_ATTR = "Print::write(data,len) should be overridden for better efficiency\r\n";
+    static bool once = false;
+    if (!once) {
+        once = true;
+        os_printf_plus(not_the_best_way);
+    }
+#endif
+
     size_t n = 0;
     while (size--) {
         size_t ret = write(*buffer++);
-        if (ret == 0) {
+        if (ret <= 0) {
```

Comment:
ret is declared of type size_t, which should be unsigned, which means it can't be negative.

Reply:
Yes very true. Blinded with ::read, arduino's api should all get a major reshape for coherency.